examples/go/tagging/handler/handler.go

```diff
@@ -31,140 +32,145 @@ type vcConfig struct {
-	moRef.Type = event.Event.Vm.Vm.Type
-
-	// Connect to vcenter with govmomi client
-	log.Println("connect to vCenter")
-	clients := connect(ctx, vcc.VCenter)
-	log.Println("connection successful")
-
-	// tag the VM moRef with tagID (URN)
-	clients.tagVM(ctx, moRef, vcc.Tag.Urn)
-
 	// Respond with a message that tagging of vm was complete
 	message := fmt.Sprintf(
 		"%v was tagged with %v",
 		moRef.Value,
-		vcc.Tag.Urn,
+		cfg.Tag.Urn,
 	)
 	log.Println(message)
 
-	err = clients.Govmomi.Logout(ctx)
-	if err != nil {
-		log.Printf("unable to log out of govmomi due to %v", err)
-	}
-	err = clients.Rest.Logout(ctx)
-	if err != nil {
-		log.Printf("unable to log out of rest due to %v", err)
-	}
-
 	return handler.Response{
 		Body:       []byte(message),
 		StatusCode: http.StatusOK,
 	}, nil
 }
 
-func loadTomlCfg() (vcConfig, error) {
-	secretPath := "/var/openfaas/secrets/vcconfig"
-	secret, err := toml.LoadFile(secretPath)
+// vsConnect connects to vSphere govmomi API using information from vcconfig.toml.
+func vsConnect() error {
+	var (
+		err        error
+		configPath = "/var/openfaas/secrets/vcconfig"
+	)
+
+	// Retrieve vCenter and tag info from the vcconfig file.
+	log.Println("load vcconfig.toml")
+	cfg, err = loadTomlCfg(configPath)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("vcconfig.toml loading failed: %w", err)
 	}
 
-	cfg := vcConfig{}
-	secret.Unmarshal(&cfg)
+	u := url.URL{
+		Scheme: "https",
+		Host:   cfg.VCenter.Server,
+		Path:   "sdk",
+	}
+	u.User = url.UserPassword(cfg.VCenter.User, cfg.VCenter.Password)
 
-	return cfg, nil
-}
+	// Currently, VEBA does not support signed certificates.
+	insecure := true
 
-// Clients needed to access vSphere APIs
-type Clients struct {
-	Govmomi *govmomi.Client
-	Rest    *rest.Client
+	log.Println("connect to vSphere")
+	client, err = vsClient.New(ctx, u, insecure)
+	if err != nil {
+		return fmt.Errorf("connection to vSphere API failed: %w", err)
+	}
+
+	return nil
 }
 
-// Connect saves connection information to clients.
-func connect(ctx context.Context, vca vCenterAuth) *Clients {
-	var c Clients
+func loadTomlCfg(path string) (vcConfig, error) {
+	cfg := vcConfig{}
 
-	u := url.URL{
-		Scheme: "https",
-		Host:   vca.Server,
-		Path:   "sdk",
+	secret, err := toml.LoadFile(path)
+	if err != nil {
+		return cfg, fmt.Errorf("unable to load vcconfig.toml: %w", err)
 	}
 
-	govc, err := govmomi.NewClient(ctx, &u, true)
+	err = secret.Unmarshal(&cfg)
 	if err != nil {
-		log.Println(err)
+		return cfg, fmt.Errorf("unable to unmarshal vcconfig.toml: %w", err)
 	}
-	c.Govmomi = govc
 
-	u.User = url.UserPassword(vca.User, vca.Password)
+	return cfg, nil
+}
 
-	err = c.Govmomi.Login(ctx, u.User)
+func parseEventMoRef(req []byte) (types.ManagedObjectReference, error) {
+	var event incoming
+	var moRef types.ManagedObjectReference
+
+	err := json.Unmarshal(req, &event)
 	if err != nil {
-		log.Print(err)
+		return moRef, fmt.Errorf("parsing of request failed: %w", err)
```

Comment:
should be `return nil, error...`

Reply:
I tried it, but `nil` is not the correct return type. The return expects a type.ManagedObjectReference type. Here's a simplified version on [Playground](https://play.golang.org/p/PTt3BZPR1VM). It produces this error: 
```
./prog.go:17:2: cannot use nil as type ManagedObjectReference in return argument
```

Reply:
I changed it such that `parseEventMoRef()` function returns a pointer to `type.MangedObjectReference`, just so I can use `nil` in case of error.